### PR TITLE
 Return accurate trophy disk space requirements

### PR DESCRIPTION
**Pull request by VelocityRa**
Makes `sceNpTrophyGetRequiredDiskSpace` return accurate space requirements and handles special case where trophies are already installed.

~*Edit:* Also sneaked in a minor unrelated commit.~

### CODE REVIEW

File: rpcs3/Loader/TRP.cpp [hunk 1]
```diff
@@ -141,3 +151,56 @@ void TRPLoader::RenameEntry(const char *oldname, const char *newname)
...
+		{
+			RemoveEntry(target);
+		}
+	}
+	return true;
+}
+
+u64 TRPLoader::GetFileSize()
+{
+	if (!m_header.trp_file_size)
```
Comment (Marthgun) [inline rpcs3/Loader/TRP.cpp]: Personally prefer this style, but every other if statement is wrapped in the code.

**Comment by Asinin3** [comment]
Hasbro Family Game Night 3 is still giving this error:
`U {PPU[0x1000006] Thread (Trophies) [0x00024ab8]} sceNpTrophy TODO: sceNpTrophyGetRequiredDiskSpace(context=0x1, handle=0x1, reqspace=*0x20ec8080, options=0x0)`
Here's the log incase it's useful.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1492047/RPCS3.log.gz)

**Comment by Megamouse** [comment]
it's not an error

**Comment by Asinin3** [comment]
Well, I know it's a TODO, but it still seems like it's what the game needs to be implemented in order to progress. But shouldn't this PR not have that TODO in the log?

**Comment by Megamouse** [comment]
the todo only tells you that the sceNpTrophyGetRequiredDiskSpace has been called.
if it failed or not is a different story

**Comment by Farseer2** [comment]
Perhaps switch it to warning, as we aren't sure this is the correct implementation yet?

**Comment by VelocityRa** [comment]
Switched it to a warning and took care of @Marthgun's code style comment.

**Comment by Nekotekina** [comment]
Don't need to change SPUAnalyser at this point.
I don't follow other changes, let's keep them at minimum.

**Comment by VelocityRa** [comment]
@Nekotekina Removed the SPUAnalyzer commit.
About the other thing, check out the separate commits, it'll make more sense.
Other than the 'Return accurate trophy disk space requirements', I moved a bunch of logic for trophy entry trimming to `TRPLoader` because it made more sense to do that there instead of `sceNpTrophy.cpp`.

**Comment by VelocityRa** [comment]
Been 2 weeks, any plans to merge this?

**Comment by Nekotekina** [comment]
Maybe, but looks very problematic. Dereferencing TRPLoader doesn't make sense.

**Comment by VelocityRa** [comment]
Oh yeah, I did that so I wouldn't have to change usages of `trp` in the following code, figured the less existing code I touched the better. I'll fix it in a few hours.
Is that the only issue?

**Comment by Gamedev1909** [comment]
Good news it no longer stalls on installing trophy data now checking trophy data 

`E {PPU[0x100001d] Thread () [0x0028b888]} sceNpTrophy: sceNpTrophyRegisterContext(context=0x1, handle=0x1, statusCb=*0xe80d50, arg=*0x216ca520, options=0x1)`

 star ocean 4 great work keep it up

**Comment by VelocityRa** [comment]
@Gamedev1909 Didn't change anything that would affect game progress in the latest push, just took care of Neko's comment.
@Nekotekina Anything else?

**Comment by VelocityRa** [comment]
_Bump..._

**Pull request closed without merging**